### PR TITLE
adding open and @objc to navigationController ivar of searchBar.

### PR DESCRIPTION
**Pull request by sdeepugd**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes
PR contains changes to make navigationcontroller ivar of searchbar open and @objc.
Reasons for this change:
 To set the navigationController of the searchbar dynamically. Right now the navigation controller of the searchbar is being set only willShow of MSFNavigationController. When the searchbar is added to the view after the ViewController is presented there is no way to set the navigationController of the searchbar. So making it public and adding @objc support to
navigationcontroller ivar of searchbar.

Binary change:

| File | Before | After | Delta |
|------|--------|-------|-------|
| libFluentUI.a | 5,59,99,600 bytes | 5,60,02,808 bytes | 3,208 bytes |

(a summary of the changes made, often organized by file)

### Verification

(how the change was tested, including both manual and automated tests)

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [x] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1440)

### CODE REVIEW

**Comment by sophialee0416** [comment]
Hi @sdeepugd, thank you for your contribution! I was wondering why you'd like to expose the navigationController property? Currently it is meant to be used as an internal variable.

**Comment by sdeepugd** [comment]
> Hi @sdeepugd, thank you for your contribution! I was wondering why you'd like to expose the navigationController property? Currently it is meant to be used as an internal variable.

Hi @sophialee0416 
In my case I am using the search button instead of the searchbar. on tapping of the searchbutton the searchbar should appear and it should disappear on tapping the cancel button. So, I will set the searchbar as the accessoryview in the target of the searchbutton.
Right now, the navigation controller of search bar is automatically set in the will show method of MSFNavigationController.
So, when I set/unset the searchbar after the will show of the MSFNavigationController there is no way the navigationcontroller of the searchbar is set externally.
That is where I need it to access it publicly to set it to current navigation controller whenever I set the searchbar dynamically.

File: ios/FluentUI/Navigation/SearchBar.swift [hunk 1]
```diff
@@ -189,7 +189,7 @@ open class SearchBar: UIView {
 
     @objc open weak var delegate: SearchBarDelegate?
 
-    weak var navigationController: NavigationController?
+	@objc open weak var navigationController: NavigationController?
```
[the remark this answers is not in the document]
Comment (sdeepugd) [reply ios/FluentUI/Navigation/SearchBar.swift:192]: Yes, it can be just public.

[the remark this answers is not in the document]
Comment (sdeepugd) [reply ios/FluentUI/Navigation/SearchBar.swift]: changed the  access specifier of nav controller from open to public

**Comment by mischreiber** [comment]
Personally I'd rather not have to expose the navigation controller -- this feels like a workaround for a behavior that we should be handling internally in Fluent.

Is there any way that we could utilize a `didSet` for the search field and fetch/set the navigation controller at that point? That way it always automatically gets the current nav controller if it's been set later.

Feel free to share a reference/link to your calling code offline if you think that will help us understand the issue.

**Comment by harrieshin** [comment]
> Personally I'd rather not have to expose the navigation controller -- this feels like a workaround for a behavior that we should be handling internally in Fluent.
> 
> Is there any way that we could utilize a `didSet` for the search field and fetch/set the navigation controller at that point? That way it always automatically gets the current nav controller if it's been set later.
> 
> Feel free to share a reference/link to your calling code offline if you think that will help us understand the issue.

has there been exploration on this direction?

**Pull request closed without merging**